anms-core/anms/routes/ARIs/reports.py

```diff
@@ -101,20 +101,25 @@ async def report_ac(agent_id: str, correlator_nonce: int):
     agent_id = agent_id.strip()
     final_res = []
     ari = None
+    dec = ace.ari_cbor.Decoder()
     # Load in adms 
     # get command that made the report as first entry 
     stmt = select(ExecutionSet).where(and_(ExecutionSet.agent_id == agent_id, ExecutionSet.correlator_nonce == correlator_nonce) )
     async with get_async_session() as session:
         result: Result = await session.scalars(stmt)
-        result = result.one_or_none()
+        
+        # there should only be one execution per agent per correlator_nonce
```

Comment:
It's also possible to trigger this by simply using an EXECSET with multiple targets, each being a CTRL reference that produces its own RPTSET upon execution finish.

Reply:
Yes that is expected behavior the agent is not required to coalesce the RPTSET together. But each report will have a different source value.